### PR TITLE
chore: Remove usage of binary/ies to describe engine(s) (files)

### PR DESCRIPTION
**Pull request by janpio**

### CODE REVIEW

File: packages/fetch-engine/src/download.ts [hunk 1]
```diff
@@ -95,30 +98,30 @@ export async function download(options: DownloadOptions): Promise<BinaryPaths> {
...
         targetFilePath,
-        envVarPath: getBinaryEnvVarPath(binaryName),
+        envVarPath: getEngineEnvVarPath(engineName),
       }
     }),
   )
 
   if (process.env.BINARY_DOWNLOAD_VERSION) {
-    opts.version = process.env.BINARY_DOWNLOAD_VERSION
+    opts.version = process.env.BINARY_DOWNLOAD_VERSION // TODO rename env var
```
Comment (Jolg42) [inline packages/fetch-engine/src/download.ts:124]:
```suggestion
    opts.version = process.env.BINARY_DOWNLOAD_VERSION // TODO rename env var to PRISMA_ENGINE_DOWNLOAD_VERSION 
```

File: packages/fetch-engine/src/download.ts [hunk 2]
```diff
@@ -455,9 +459,10 @@ async function saveFileToCache(
   }
 }
 
-function engineTypeToBinaryType(engineType: string, binaryTarget: string): string {
-  if (BinaryType[engineType]) {
-    return BinaryType[engineType]
+// TODO ???? What is this doing?
+function barToBaz(engineType: string, binaryTarget: string): string {
```
Comment (Jolg42) [inline packages/fetch-engine/src/download.ts:463]: Looks like it's returning the "file name" of the engine, and it handles "native" 
@janpio Can you revert this rename? "barToBaz"

File: packages/internals/src/__tests__/getGenerators/getGenerators.test.ts [hunk 3]
```diff
@@ -578,21 +578,21 @@ describe('getGenerators', () => {
...
       providerAliases: aliases,
-      binaryPathsOverride: {
+      enginePathsOverride: {
         queryEngine: queryEnginePath,
       },
       dataProxy: false,
     })
 
-    const options = generators.map((g) => g.options?.binaryPaths)
+    const options = generators.map((g) => g.options?.binaryPaths) // TODO enginePaths
```
Comment (Jolg42) [inline packages/internals/src/__tests__/getGenerators/getGenerators.test.ts:595]:
```suggestion
    const options = generators.map((g) => g.options?.binaryPaths) // TODO rename binaryPaths to enginePaths
```

File: packages/client/src/runtime/getPrismaClient.ts [hunk 4]
```diff
@@ -130,7 +130,7 @@ export interface PrismaClientOptions {
     hooks?: Hooks
     engine?: {
       cwd?: string
-      binaryPath?: string
+      binaryPath?: string // TODO Rename
```
Comment (Jolg42) [inline packages/client/src/runtime/getPrismaClient.ts:133]:
```suggestion
      binaryPath?: string // TODO Rename to enginePath
```

File: packages/engine-core/src/binary/BinaryEngine.ts [hunk 5]
```diff
@@ -210,6 +210,7 @@ export class BinaryEngine extends Engine {
 You may have to run ${chalk.greenBright('prisma generate')} for your changes to take effect.`,
           this.clientVersion!,
         )
+        // TODO Replace `binary` here and in snapshots
```
Comment (Jolg42) [inline packages/engine-core/src/binary/BinaryEngine.ts:213]:
```suggestion
        // TODO Replace `binary` by `engine` here and in snapshots
```

File: packages/engine-core/src/binary/BinaryEngine.ts [hunk 6]
```diff
@@ -375,7 +376,7 @@ You may have to run ${chalk.greenBright('prisma generate')} for your changes to
         ? `\nYou incorrectly pinned it to ${chalk.redBright.bold(`${this.incorrectlyPinnedBinaryTarget}`)}\n`
         : ''
 
-      let errorText = `Query engine binary for current platform "${chalk.bold(
+      let errorText = `Query engine binary for current platform "${chalk.bold( // TODO replace `binary` here and in snapshots
```
Comment (Jolg42) [inline packages/engine-core/src/binary/BinaryEngine.ts:379]:
```suggestion
      let errorText = `Query engine binary for current platform "${chalk.bold( // TODO replace `binary` by `engine here and in snapshots
```

File: packages/engine-core/src/binary/BinaryEngine.ts [hunk 7]
```diff
@@ -649,7 +650,7 @@ ${chalk.dim("In case we're mistaken, please report this to us 🙏.")}`)
             } else if (this.child?.signalCode) {
               err = new PrismaClientInitializationError(
                 `Query engine process killed with signal ${this.child.signalCode} for unknown reason.
-Make sure that the engine binary at ${prismaPath} is not corrupt.\n` + msg,
+Make sure that the engine binary at ${prismaPath} is not corrupt.\n` + msg, // TODO replace `binary` here and in snapshots
```
Comment (Jolg42) [inline packages/engine-core/src/binary/BinaryEngine.ts:653]:
```suggestion
Make sure that the engine binary at ${prismaPath} is not corrupt.\n` + msg, // TODO replace `binary` by `engine here and in snapshots
```

File: packages/fetch-engine/src/__tests__/download.test.ts [hunk 8]
```diff
@@ -106,50 +106,50 @@ describe('download', () => {
...
-      binaries: {
+      engines: {
         'query-engine': baseDir,
       },
-      version: FIXED_BINARIES_HASH,
+      version: FIXED_ENGINES_HASH,
     })
 
-    fs.writeFileSync(targetPath, 'incorrect-binary')
+    fs.writeFileSync(targetPath, 'incorrect-binary') // TODO
```
Comment (Jolg42) [inline packages/fetch-engine/src/__tests__/download.test.ts:115]:
```suggestion
    fs.writeFileSync(targetPath, 'incorrect-binary') // TODO rename `binary` to `engine`
```

File: packages/generator-helper/src/types.ts [hunk 9]
```diff
@@ -88,7 +88,7 @@ export type GeneratorOptions = {
   datamodel: string
   // TODO is it really always version hash? Feature is unclear.
   version: string // version hash
-  binaryPaths?: BinaryPaths
+  binaryPaths?: EnginePaths // TODO Rename after Studio was adapted - currently it depends on this
```
Comment (Jolg42) [inline packages/generator-helper/src/types.ts:91]:
```suggestion
  binaryPaths?: EnginePaths // TODO Rename `binaryPaths` to `enginePaths` after Studio was adapted - currently it depends on this
```

File: packages/internals/src/get-generators/utils/binaryTypeToEngineType.ts [hunk 10]
```diff
@@ -1,22 +1,23 @@
-import { BinaryType } from '@prisma/fetch-engine'
+import { EngineTypeEnum } from '@prisma/fetch-engine'
 import type { EngineType } from '@prisma/generator-helper'
 
-export function binaryTypeToEngineType(binaryType: string): EngineType {
-  if (binaryType === BinaryType.introspectionEngine) {
+// TODO: ??? binaryTypeToEngineType
+export function barToFoo(binaryType: string): EngineType {
```
Comment (Jolg42) [inline packages/internals/src/get-generators/utils/binaryTypeToEngineType.ts:5]: Can you revert this rename?

**Review by Jolg42 — CHANGES_REQUESTED** [review]
Thanks! 💚 

I added some suggestions to make the TODO more explicit 

and please revert the following renaming before we can merge: barToBaz barToFoo fooToBar
We could tackle this separately, I also found out that you had a big list of things detailed in https://github.com/prisma/prisma/pull/12297

**Review by Jolg42 — CHANGES_REQUESTED** [review]
Note that we should wait for https://github.com/prisma/prisma/pull/14121 to be merged and then update this PR and see if all tests pass again.

File: packages/cli/src/Version.ts [hunk 11]
```diff
@@ -1,9 +1,9 @@
-import { enginesVersion, getCliQueryEngineBinaryType } from '@prisma/engines'
+import { enginesVersion, getCliQueryEngineType } from '@prisma/engines'
 import { getPlatform } from '@prisma/get-platform'
 import type { Command } from '@prisma/internals'
 import {
   arg,
-  BinaryType,
+  EngineTypeEnum,
```
Comment (jkomyno) [inline packages/cli/src/Version.ts:6]: Could we call it `EngineType` instead?

File: packages/cli/src/Version.ts [hunk 12]
```diff
@@ -127,22 +126,22 @@ export class Version implements Command {
...
       return { version, path: pathFromEnv, fromEnvVar: envVar }
     }
 
-    const binaryPath = await resolveBinary(binaryName)
-    const version = await getEngineVersion(binaryPath, binaryName)
-    return { path: binaryPath, version }
+    const enginePath = await resolveEngine(engineName)
+    const version = await getEngineVersion(enginePath, engineName)
+    return { path: enginePath, version }
   }
```
Comment (jkomyno) [inline packages/cli/src/Version.ts:145]: Watch out when merging, as this block here will surely create conflicts.

File: packages/client/src/runGeneratorBinary.ts [hunk 13]
```diff
@@ -1,10 +1,10 @@
 import execa from 'execa'
 
-export async function runGeneratorBinary(
-  binaryPath: string,
+export async function runGeneratorExecutable(
+  executablePath: string,
   options: any,
 ): Promise<{ stdout: string; stderr: string }> {
-  const child = execa(binaryPath)
+  const child = execa(executablePath)
```
Comment (jkomyno) [inline packages/client/src/runGeneratorBinary.ts:7]: Here we can only execute engine binaries, right? If so, probably we can avoid renaming the function and the argument

Comment (janpio) [reply packages/client/src/runGeneratorBinary.ts:7]: No, we can execute anything executable.

**Pull request closed without merging**